index.d.ts

```diff
@@ -17,9 +17,10 @@ import {
 } from "graphql";
 import { SocketStream } from "fastify-websocket"
 import { IncomingMessage, IncomingHttpHeaders, OutgoingHttpHeaders } from "http";
+import { Readable } from "stream";
 
 export interface PubSub {
-  subscribe<TResult = any>(topics: string | string[]): Promise<AsyncIterator<TResult>>;
+  subscribe(topics: string | string[]): Promise<Readable>;
```

Comment:
Most people does not know that a Readable is an asyncIterator :/.

Reply:
I was playing with the types and it seems like `Readable & AsyncIterable<TResult>` seems to be a good middle-ground, while allowing to access the readable functions & keeping the async iterable stuff with the data type

https://codesandbox.io/s/summer-butterfly-xgxdv?file=/src/index.ts

Reply:
Agree!